src/Data/Equality/Graph/Internal.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE UndecidableInstances #-} -- tmp show
+{-# LANGUAGE FlexibleContexts #-}
```

Comment:
```suggestion
{-# LANGUAGE UndecidableInstances #-} -- tmp show
{-# LANGUAGE FlexibleContexts #-}
{-# OPTIONS_HADDOCK hide #-}
```

This will prevent the Haddock documentation, on Hackage, from having a link to this module. The module still appears in the module list, but it's visually distinct. I recommend using this for all internal modules. Like this:

![haddock hide](https://user-images.githubusercontent.com/1413217/187066404-6b0515dd-a065-4f1d-b7ae-87bfc9438602.png)

.

Reply:
I'm currently not exporting the module since it's listed in `other-modules` rather than `exposed-modules`.

I'm not convinced the extra noise in the haddock modules list is of any benefit.

Reply:
Ah, yes, I tend to forget about `other-modules` because I don't use it. It's a double-edged sword (it used to be bad, because modules in the `other-modules` category didn't even get their code on haddock, so some exposed functions would not have source code and it was quite unpleasant an experience).

It's a matter of taste, I guess. I've just been bitten too many times do to something being tucked in other-modules so left inaccessible to me, but I'd need it for a workaround waiting for something to be fixed upstream (sometimes, it's worth it to access the internal representation of something to do some ugly hack). I've also had types exported from some `other-modules` that appeared in the interface of libraries, and so there were functions whose type I couldn't write. So I decided long ago to never take this sort of risk for my users, and not use `other-modules`. But that's me :slightly_smiling_face: .